.pre-commit-config.yaml

```
@@ -0,0 +1,31 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+    -   id: check-yaml
```

There's a few more hooks that would be good to enable:
```suggestion
      - id: check-added-large-files
      - id: check-merge-conflict
      - id: check-yaml
      - id: detect-private-key
      - id: end-of-file-fixer
      - id: trailing-whitespace
```

---

Are not `trailing-whitespace` covered by [Ruff rule W291](https://docs.astral.sh/ruff/rules/trailing-whitespace/) and `end-of-file-fixer` by [Ruff rule W292](https://docs.astral.sh/ruff/rules/missing-newline-at-end-of-file/)?

---

Ruff is a Python linter, it doesn't check all files.